students/tfbanks/Lesson05/Mailroom_pt3.py

```diff
@@ -0,0 +1,89 @@
+# Mailroom_pt3.py Exercise by tfbanks
+
+# !/usr/bin/env python3
+
+
+# Donor Database containing names and donations
+donor_names = {
+    'Tim Cooker': [1500.50, 25050.50, 15680.75],
+    'Elon Musket': [5250.25, 26500],
+    'Frank Petersmankempt': [550.60],
+    'Megan Morgan': [650.40, 1600, 20625.40],
+    'Marlene Wheeler': [820, 1222.80]}
+
+
+def thank_you():  # Code for selecting Donor and writing a Thank You Note
+    ty_donor = input('Enter the Full Name of a donor (for list of previous donors type list): ').title()
+    letter = ('''\n\nDear {},\n
+Thank you for your generous donation of ${:,.2f}, your generosity is greatly appreciated.
+These funds will help continue our efforts to teach Python to the next generation.\n
+Sincerely,\n\n
+Joesef Edword Bringingham\n\n''')
+    if ty_donor == 'List':  # If List, Prints Donor List and restarts the function
+        for donor in donor_names.keys():
+            print(donor)
+        print('\n')
+        ty_donor = input('Enter the Full Name of a donor (for list of previous donors type list): ').title()
+
+    while True:
+        try:
+            amount = float(input('Please enter the donation amount: '))
+        except ValueError:
+            print('Please enter a valid donation amount')
+            continue
+        break
+
+    if ty_donor in donor_names:  # Actions if donor is already in donor_names
+        donor_names[ty_donor].append(amount)  # Appends donation amount to donators donation values
+    else:
+        donor_names[ty_donor] = [amount]
+
+    print(letter.format(ty_donor, amount))
+
+
+def report():  # Defines parameters of the report of donors, creates a new list of summarized details per donor, sorts, and prints it.
+    print('\nDonor Name            |  Total Given | Num Gifts |  Average Gift')
+    print('------------------------------------------------------------------')
+    donor_rpt = [[donor, sum(donations), len(donations), sum(donations)/len(donations)] for donor, donations in donor_names.items()]
```

Comment:
Usually you want to avoid one letter vars, or short vars but in the case of comprehensions it's ok.

Very common pattern is to use `for k, v in some_dict.items()` or `for key, val in some_dict.items()`
so that the comprehension line is not overly long.